ports/qemu-riscv/README.md

```diff
@@ -0,0 +1,71 @@
+This is an experimental, community-supported port for RISC-V RV32IMC emulation
+as provided by QEMU (http://qemu.org).
+
+The purposes of this port are to enable:
+
+1. Continuous integration
+    - run tests against architecture-specific parts of code base
+2. Experimentation
+    - simulation & prototyping of anything that has architecture-specific
+      code
+    - exploring instruction set in terms of optimising some part of
+      MicroPython or a module
+3. Streamlined debugging
+    - no need for JTAG or even an MCU chip itself
+    - no need to use OpenOCD or anything else that might slow down the
+      process in terms of plugging things together, pressing buttons, etc.
+
+This port requires a bare metal RISC-V toolchain with GCC 10 or later, either
+with multilib support or 32 bits specific (M, C, and Zicsr extensions must be
+supported, along with ilp32 ABI).  Both newlib and picolibc are supported,
+with the latter having precedence if found.
+
+Most pre-built toolchains should work out of the box, either coming from your
+Linux distribution's package manager, or independently packaged ones like
+[xPack](https://xpack.github.io/dev-tools/riscv-none-elf-gcc/).
+
+To build and run the image with builtin testsuite:
+
+    make -f Makefile.test test
+
+If that fails, make sure your toolchain satisfies the given requirements.  If
+it does, try to check whether the values of `ARCH` and `ARCH_LD` are what your
+toolchain expects.  If it does not, you may want to pick a supported toolchain,
+or you can build your own following the instructions on
```

Comment:
I appreciate you went to a lot of work here to document how to build a custom toolchain, however I kind of feel like it's better to not include this entire section and encourage everyone to use the pre-built ones now they're working OK. 

Of course if someone wants to build their own toolchain then they can, but it feels like "here be monsters" territory to me so better left for people to adventure on their own. :grin:

Reply:
I thought a bit about it, and I think you're right here.  I originally put it there to avoid having folks wondering how to get that working, but I now realise pretty much nobody who knows how to build their own toolchain (or knows where to look for pointers) would ever try to run this in the first place.

Anyway, the section was taken out of the file.